tests/core/engine_adapter/test_integration.py

```diff
@@ -96,6 +101,20 @@ def dialect(self) -> str:
     def current_catalog_type(self) -> str:
         return self.engine_adapter.current_catalog_type
 
+    @property
+    def supports_merge(self) -> bool:
```

Comment:
I found that this needed to be checked in multiple places so I made it a property of the TestContext

In addition, it's become more complex. It's no longer "engine either supports merge or it doesnt", it depends on both the engine _and_ the catalog

Reply:
Interesting. What prompted you to change this? Did you run into any issues? Or did you notice that merge is not being used?

Reply:
I noticed that some other tests, like `test_merge()` werent actually testing all the combinations because they were being skipped where `dialect=spark` or `dialect=trino`.

Spark and Trino _do_ support MERGE, just not on all catalogs, so I improved the check.

But, I wanted to use the same logic in the ` test_batch_size_on_incremental_by_unique_key_model` test (because unless i'm missing something, `INCREMENTAL_BY_UNIQUE_KEY` only works on engines that support `MERGE`). So rather than duplicate the checks, I made them a property of the `TestContext` fixture so they could be used in both places.

Arguably this kind of thing could also live on the EngineAdapter